modules/pam_ftp/pam_ftp.c

```diff
@@ -133,6 +133,11 @@ pam_sm_authenticate (pam_handle_t *pamh, int flags UNUSED,
 	retval = pam_set_item(pamh, PAM_USER, (const void *)anon_user);
 	if (retval != PAM_SUCCESS || anon_user == NULL) {
 	    pam_syslog(pamh, LOG_ERR, "user resetting failed");
+
+        if (anon_user != NULL) {
+            free(anon_user);
+        }
```

Comment:
There is no need to check for NULL before free().

Reply:
Also the indentation is off.

Reply:
Updated